### PR TITLE
Clean up Pytorch lightning GMM feature branch

### PR DESCRIPTION
**Pull request by Ianlmgoddard**
This PR remove references to different covariance types as the functionality for covariance types other than "full" has not been verified.  Additionally, reference to KMeans Lightning are removed, as we will use sklearn kmeans for now.

### CODE REVIEW

File: src/opensynth/models/faraday/gaussian_mixture/fit_kmeans.py [hunk 1]
```diff
@@ -33,36 +33,6 @@ def fit_kmeans(
...
-    # trainer = pl.Trainer(
-    #     max_epochs=max_epochs,
-    #     accelerator=accelerator,
-    #     devices=devices,
-    #     logger=logger,
-    # )
-    # trainer.fit(kmeans_module, data)
-
     # TODO : perform k-means on a random subsample of the training dataset to
     # speed up convergence. This is necessary when working with large datasets.
```
Comment (charlotte-avery) [inline src/opensynth/models/faraday/gaussian_mixture/fit_kmeans.py:37]: Can you please add a note to say that k-means fitting is done on the first batch in the DataLoader (see `next_batch=next(iter(data))`). We can probably remove the TODO statement given that this first batch is a random subsample of the dataset (provided the dataset is shuffled either in the DataModule or in the database from which we are calling, which it is).

File: src/opensynth/models/faraday/gaussian_mixture/model.py [hunk 2]
```diff
@@ -245,8 +191,6 @@ def sample(self, num_datapoints: int) -> torch.Tensor:
         return torch.cat(result, dim=0)
 
     def _get_component_precision(self, component: int) -> torch.Tensor:
-        if self.covariance_type == "tied":
-            return self.precisions_cholesky
         return self.precisions_cholesky[component]
```
Comment (charlotte-avery) [inline src/opensynth/models/faraday/gaussian_mixture/model.py:196]: K-means model is redundant (since we are now using the sk-learn k-means) so you can delete the code below this line in `model.py`. Also might want to rename this script to `gmm_model.py` to not confuse it with the `model.py` script in the parent directory?

File: src/opensynth/models/faraday/gaussian_mixture/metrics.py [hunk 3]
```diff
@@ -143,56 +132,22 @@ def update(
...
-            result[:: self.num_features + 1].add_(self.reg)
-            return result.view(shape)
+            covars_regularized = covars + regularization
+
+            self.covariance_sum[i].add_(covars_regularized)
+
+    def compute(self) -> torch.Tensor:
         # covariance_type == "full"
         result = self.covariance_sum / self.component_weights.unsqueeze(
             -1
```
Comment (charlotte-avery) [inline src/opensynth/models/faraday/gaussian_mixture/metrics.py:153]: Remove k-means metrics (line 168 onwards) as they are no longer needed for sk-learn implementation.

File: src/opensynth/models/faraday/gaussian_mixture/fit_kmeans.py [hunk 4]
```diff
@@ -33,36 +33,6 @@ def fit_kmeans(
...
     kmeans_model_ = KMeans(n_clusters=num_components)
```
Comment (charlotte-avery) [inline src/opensynth/models/faraday/gaussian_mixture/fit_kmeans.py:38]: Lines 40-44 can be replaced by calling the function `encode_data_for_gmm` from `prepare_gmm_input.py`. (Except you would still have to  `.detach().numpy()` here as this is not done in  `encode_data_for_gmm`). Note, do not use the `prepare_data_for_model` function as this will expand the dataset based on the weights which we do not want to do for k-means training (only GMM).

File: src/opensynth/models/faraday/gaussian_mixture/model.py [hunk 5]
```diff
@@ -68,18 +67,11 @@ def covariances(self) -> torch.Tensor:
         """
         The covariance matrices learnt for the GMM's components.
 
-        The shape of the tensor depends on the covariance type,
-            see :class:`CovarianceType`.
+        The shape of the tensor depends is
```
Comment (charlotte-avery) [inline src/opensynth/models/faraday/gaussian_mixture/model.py:70]: Remove 'depends'

**Review by charlotte-avery — CHANGES_REQUESTED** [review]

**Review by charlotte-avery — APPROVED** [review]
lgtm, thanks Ian!
